pkg/reconciler/events/cloudevent/cloudevent.go

```diff
@@ -71,27 +118,62 @@ func EventForTaskRun(taskRun *v1beta1.TaskRun) (*cloudevents.Event, error) {
 	if taskRun == nil {
 		return nil, errors.New("Cannot send an event for an empty TaskRun")
 	}
-	event := cloudevents.NewEvent()
-	event.SetID(uuid.New().String())
-	event.SetSubject(taskRun.ObjectMeta.Name)
-	event.SetSource(taskRun.ObjectMeta.SelfLink) // TODO: SelfLink is deprecated
+	return EventForObjectWithCondition(taskRun)
+}
+
+// EventForPipelineRun will create a new event based on a TaskRun,
+// or return an error if not possible.
+func EventForPipelineRun(pipelineRun *v1beta1.PipelineRun) (*cloudevents.Event, error) {
+	// Check if the TaskRun is defined
+	if pipelineRun == nil {
+		return nil, errors.New("Cannot send an event for an empty PipelineRun")
+	}
+	return EventForObjectWithCondition(pipelineRun)
+}
 
-	c := taskRun.Status.GetCondition(apis.ConditionSucceeded)
+func getEventType(runObject objectWithCondition) (*TektonEventType, error) {
+	c := runObject.GetStatusCondition().GetCondition(apis.ConditionSucceeded)
+	var eventType TektonEventType
 	switch {
 	case c.IsUnknown():
-		event.SetType(TektonTaskRunUnknownV1.String())
+		switch runObject.(type) {
+		case *v1beta1.TaskRun:
+			switch c.Reason {
+			case v1beta1.TaskRunSucceededReasonStarted.String():
+				eventType = TektonTaskRunStartedV1
+			case v1beta1.TaskRunSucceededReasonRunning.String():
+				eventType = TektonTaskRunRunningV1
+			default:
+				eventType = TektonTaskRunUnknownV1
+			}
+		case *v1beta1.PipelineRun:
+			switch c.Reason {
+			case v1beta1.PipelineRunSucceededReasonStarted.String():
+				eventType = TektonPipelineRunStartedV1
+			case v1beta1.PipelineRunSucceededReasonRunning.String():
+				eventType = TektonPipelineRunRunningV1
+			default:
+				eventType = TektonPipelineRunUnknownV1
+			}
+		}
 	case c.IsFalse():
-		event.SetType(TektonTaskRunFailedV1.String())
+		switch runObject.(type) {
+		case *v1beta1.TaskRun:
+			eventType = TektonTaskRunFailedV1
+		case *v1beta1.PipelineRun:
+			eventType = TektonPipelineRunFailedV1
+		}
 	case c.IsTrue():
-		event.SetType(TektonTaskRunSuccessfulV1.String())
+		switch runObject.(type) {
+		case *v1beta1.TaskRun:
+			eventType = TektonTaskRunSuccessfulV1
+		case *v1beta1.PipelineRun:
+			eventType = TektonPipelineRunSuccessfulV1
+		}
 	default:
 		return nil, fmt.Errorf("unknown condition for in TaskRun.Status %s", c.Status)
```

Comment:
NIT: we are now including PipelineRun.Status here as well, may be change this error message to include PipelineRun as well

Reply:
done